dependencies/clients/teku.go

```diff
@@ -0,0 +1,329 @@
+package clients
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+
+	"github.com/lukso-network/tools-lukso-cli/common/errors"
+	"github.com/lukso-network/tools-lukso-cli/common/system"
+	"github.com/lukso-network/tools-lukso-cli/common/utils"
+	"github.com/lukso-network/tools-lukso-cli/flags"
+	"github.com/lukso-network/tools-lukso-cli/pid"
+)
+
+const (
+	tekuDepsFolder = "teku" // folder in which both teku and JDK are stored
+	tekuFolder     = "teku" // folder in which teku is stored (in tekuDepsFolder)
+	jdkFolder      = "jdk"  // folder in which JDK is stored (in tekuDepsFolder)
+	jdkInstallURL  = "https://download.java.net/java/GA/jdk20.0.2/6e380f22cbe7469fa75fb448bd903d8e/9/GPL/openjdk-20.0.2_linux-x64_bin.tar.gz"
```

Comment:
we should stick to the version used by Teku, which is 17.

Reply:
`Building on a more recent version of Java is supported, but the resulting build will not work on earlier versions of Java.`
Honestly I think that this 20 version is fine - jdk.java.net doesn't even support version 17 at the moment